File: atcoder/abc126B_yymm_or_mmyy.py
# Vicfred
# https://atcoder.jp/contests/abc126/tasks/abc126_b
# implementation
s = int(input())

yy = s // 100
mm = s % 100

if 1 <= yy <= 12:
    if 1 <= mm <= 12:
        print("AMBIGUOUS")
    else:
        print("MMYY")
else:
    if 1 <= mm <= 12:
        print("YYMM")
    else:
        print("NA")

File: atcoder/abc126B_yymm_or_mmyy.ml
(* Vicfred
 * https://atcoder.jp/contests/abc126/tasks/abc126_b
 * implementation
 * *)
let a, b = Scanf.scanf " %2d%2d" @@ fun a b -> a, b
let f x = x = 0 || x > 12
let main = print_endline @@
match f a, f b with
true, true -> "NA" |
true, false -> "YYMM" |
false, true -> "MMYY" |
_ -> "AMBIGUOUS"

